### PR TITLE
Fix sql

### PR DESCRIPTION
**Pull request by da-kami**
That was a tricky one 😅

This should fix all issues where we get an error complaining that we try to transition to `Accepted` even though the CFD was already accepted (which it was not...).
This problem might be the source of several weird side effects.

### CODE REVIEW

File: daemon/src/db.rs [hunk 1]
```diff
@@ -472,6 +473,56 @@ mod tests {
...
+
+        let order_id = order.id;
+
+        insert_order(&order, &mut conn).await.unwrap();
+        insert_cfd(cfd.clone(), &mut conn).await.unwrap();
+
+        let cfd_from_db = load_cfd_by_order_id(order_id, &mut conn).await.unwrap();
+        assert_eq!(cfd, cfd_from_db);
+
+        // insert second cfd with order-id that is higher in alphabet
```
Comment (thomaseizinger) [inline daemon/src/db.rs]: You gotta be joking ...

**Review by thomaseizinger — APPROVED** [review]
Thank you for the elaborate commit message, that made things really easy to understand :)

File: daemon/src/db.rs (hunk 1 above)
Comment (da-kami) [reply daemon/src/db.rs]: when I saw your message I actually thought - yeah... that one needs more validation. 
It is actually a more general error. The UUID does not matter (I just had used the failing ones to reproduce it). I just flipped the UUIDs and it still works. Will see to generalize the test on Monday and remove the comment!

Comment (da-kami) [reply daemon/src/db.rs]: One thing that I find surprising: sqlx does not complain when `fetch_one` returns more than one.

Comment (bonomat) [reply daemon/src/db.rs]: Given `fetch_one` does not fail, shall we better not use it then?

Comment (thomaseizinger) [reply daemon/src/db.rs]: The docs are somewhat ambiguous here. It sort of suggests that it fails if there are more than 2 results but also doesn't say that clearly anywhere.

Best to open an issue!

Not sure if we need to entirely avoid it but it is good to keep the behaviour in mind. If we do decide that we don't want it, I want it enforced via https://rust-lang.github.io/rust-clippy/rust-1.55.0/index.html#disallowed_method.

**Review by bonomat — APPROVED** [review]

File: daemon/src/db.rs (hunk 1 above)
Comment (da-kami) [reply daemon/src/db.rs]: I had another look, and to be fair the documentation states:

> Errors if no rows were returned. **Extra rows are ignored.** Aggregate queries, use this.

https://docs.rs/sqlx/0.5.1/sqlx/macro.query.html

This means we should probably avoid using it. I'd still go for merge now - we can re-visit this for refactoring later. Better get the fix in :)

Comment (klochowicz) [reply daemon/src/db.rs]: @da-kami perhaps we could go in for merge now and create an issue to remove `fetch_one()` from the code?

**Review by klochowicz — APPROVED** [review]

**Pull request merged**